The code has been introduced by commit 2cbb93a436eaa78591677ca5ac3886f94471f485.

Snip:
```java
// Create a new task manager object
try {
	new TaskManager();
} catch (Exception e) {
	LOG.fatal("Taskmanager startup failed: " + e.getMessage(), e);
	System.exit(FAILURE_RETURN_CODE);
}

// park the main thread to keep the JVM alive (all other threads may be daemon threads)
try {
	new Object().wait(); // line 421
} catch (InterruptedException ex) {}
```

The exception

```
java.lang.IllegalMonitorStateException
	at java.lang.Object.wait(Native Method)
	at java.lang.Object.wait(Object.java:502)
	at eu.stratosphere.nephele.taskmanager.TaskManager.main(TaskManager.java:421)
	at eu.stratosphere.yarn.YarnTaskManagerRunner$1.run(YarnTaskManagerRunner.java:55)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:357)
	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1471)
	at eu.stratosphere.yarn.YarnTaskManagerRunner.main(YarnTaskManagerRunner.java:51)
```

I think you can only call `.wait()` inside a synchronized block?

---------------- Imported from GitHub ----------------
Url: https://github.com/stratosphere/stratosphere/issues/739
Created by: rmetzger
Labels: bug, runtime, simple-issue, 
Created at: Tue Apr 29 18:10:18 CEST 2014
State: closed